### PR TITLE
Add `NonPreemptivePriorityScheduling` new algorithm with unit tests

### PR DESCRIPTION
**Pull request by Hardvan**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ */
+public final class NonPreemptivePriorityScheduling {
+
+    private NonPreemptivePriorityScheduling() {
+    }
+
+    static class Process implements Comparable<Process> {
+        int id; // Process ID
+        int burstTime; // Time required by the process for execution
+        int priority; // Priority of the process (lower value indicates higher priority)
```
Comment (siriak) [inline src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java:19]: Shouldn't the process also have start time? Usually in scheduling they don't start all at once

Comment (Hardvan) [reply src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java:19]: You are right, I will add the `startTime` attribute as well.

**Comment by Hardvan** [comment]
I have added the `startTime` attribute to the scheduling code and modified the Junit tests to reflect the same.

File: src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    private NonPreemptivePriorityScheduling() {
+    }
+
+    /**
+     * Represents a process with an ID, burst time, priority, and start time.
+     */
+    static class Process implements Comparable<Process> {
+        int id;
+        int startTime;
```
Comment (siriak) [inline src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java:21]: I meant not the time when the process is started, but when it arrives in the system. It appears and can be executed

Comment (Hardvan) [reply src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java:22]: I added another variable called `arrivalTime` to keep track of this.

**Comment by Hardvan** [comment]
I added another variable called `arrivalTime` to keep track of this.

File: src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        Process[] executionOrder = new Process[processes.length];
+
+        for (Process process : processes) {
+            pq.add(process);
+        }
+
+        while (!pq.isEmpty()) {
+            Process currentProcess = pq.poll();
+
+            if (currentTime < currentProcess.arrivalTime) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java:79]: You just idle while waiting for the highest priority process to arrive, but should be executing other processes during that time https://www.educative.io/answers/what-is-a-non-preemptive-priority-scheduling-algorithm

Comment (Hardvan) [reply src/main/java/com/thealgorithms/scheduling/NonPreemptivePriorityScheduling.java]: Added `waitingQueue` variable to address the above change.

**Comment by Hardvan** [comment]
I have added a `waitingQueue` variable to address the above change.

**Review by siriak — APPROVED** [review]

**Pull request merged**